src/core/portal/dom-portal-host.ts

```diff
@@ -0,0 +1,56 @@
+import {DynamicComponentLoader, AppViewManager, ComponentRef} from 'angular2/core';
+import {BaseException} from 'angular2/src/facade/exceptions';
+import {BasePortalHost, ComponentPortal, TemplatePortal} from './portal';
+
+
+/**
+ * A PortalHost for attaching portals to an arbitrary DOM element outside of the Angular
+ * application context.
+ *
+ * This is the only part of the portal core that directly touches the DOM.
+ */
+export class DomPortalHost extends BasePortalHost {
+  constructor(
+      private _hostDomElement: Element,
```

Comment:
Would it be cleaner to get an `ElementRef` here? Alternatively, I'm wondering if you could get the `Renderer` and use `projectNodes()`.